File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/availabilityMenuItemAssociations/UpdateAvailabilityMenuItemAssociationsRequest.ts
export interface UpdateAvailabilityMenuItemAssociationsRequest {
    readonly availabilityId?: string;
    readonly menuItemId?: string;
    readonly associations: UpdateAvailabilityMenuItemAssociation[];
}

export interface UpdateAvailabilityMenuItemAssociation {
    readonly id: string;
    readonly active: boolean;
}

File: Quivi.Backoffice.Api/Requests/AvailabilityMenuItemAssociations/UpdateAvailabilityMenuItemAssociationsRequest.cs
namespace Quivi.Backoffice.Api.Requests.AvailabilityMenuItemAssociations
{
    public class UpdateAvailabilityMenuItemAssociationsRequest : ARequest
    {
        public required IEnumerable<UpdateAvailabilityMenuItemAssociation> Associations { get; init; }
    }

    public class UpdateAvailabilityMenuItemAssociation
    {
        public required string Id { get; init; }
        public bool Active { get; init; }
    }
}